Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp

```diff
@@ -153,12 +166,12 @@ std::optional<FloatSize> getVideoResolutionFromCaps(const GstCaps* caps)
     int width = 0, height = 0;
     int pixelAspectRatioNumerator = 1, pixelAspectRatioDenominator = 1;
 
+    GstStructure* structure = gst_caps_get_structure(caps, 0);
     if (areEncryptedCaps(caps)) {
-        GstStructure* structure = gst_caps_get_structure(caps, 0);
         gst_structure_get_int(structure, "width", &width);
         gst_structure_get_int(structure, "height", &height);
         gst_structure_get_fraction(structure, "pixel-aspect-ratio", &pixelAspectRatioNumerator, &pixelAspectRatioDenominator);
-    } else {
+    } else if (!gst_structure_has_name(structure, "video/x-raw") || gst_structure_has_name(structure, "format")) {
```

Comment:
Ditto. Did you mean to use `gst_structure_has_field()` here?

Reply:
Oh, probably.

Reply:
Changed to `gst_structure_has_field`.